src/test/java/org/prebid/server/bidder/bidmachine/BidmachineBidderTest.java

```diff
@@ -0,0 +1,274 @@
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateCorrectURL() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder.banner(Banner.builder()
+                        .format(singletonList(Format.builder().w(300).h(500).build()))
+                        .build()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidmachineBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getUri)
+                .containsExactly("https://127.0.0.1/path/1");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldNotModifyImplIfNoPrebidIsRequest() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder.banner(Banner.builder()
+                        .format(singletonList(Format.builder().w(300).h(500).build()))
+                        .build()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidmachineBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getPayload)
+                .containsExactly(bidRequest);
+    }
+
+    @Test
+    public void makeHttpRequestsShouldModifyImplIfPrebidIsRequestAndBannerBattrDoesNotContain16() {
+        // given
+        List<Integer> requestBattr = new ArrayList<>();
+        requestBattr.add(1);
+        final Imp imp = givenImp(impBuilder -> impBuilder
+                .banner(Banner.builder()
+                        .format(singletonList(Format.builder().w(300).h(500).build()))
+                        .battr(requestBattr).build())
+                .ext(mapper.valueToTree(ExtPrebid.of(
+                        ExtImpPrebid.builder()
+                                .isRewardedInventory(1)
+                                .build(), ExtImpBidmachine.of("host", "pubId", "1")))));
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(imp))
+                .build();
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidmachineBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getPayload)
+                .extracting(BidRequest::getImp)
+                .extracting(imps -> imps.get(0))
+                .extracting(currImp -> currImp.getBanner().getBattr())
+                .containsExactly(Arrays.asList(1, 16));
```

Comment:
Flat extracting and then Arrays.asList can be omitted, check for similar occurrences